lib/CXGN/Stock/Order.pm

```diff
@@ -0,0 +1,200 @@
+
+package CXGN::Stock::Order;
+
+use Moose;
+use Data::Dumper;
+use CXGN::Stock::OrderBatch;
+use CXGN::People::Person;
+use JSON;
+
+has 'people_schema' => ( isa => 'Ref', is => 'rw', required => 1 );
+
+has 'dbh' => (is  => 'rw', required => 1,);
+
+has 'sp_order_id' => (isa => 'Int', is => 'rw' );
+
+has 'order_from_id' => ( isa => 'Int', is => 'rw' );
+
+has 'order_to_id' => ( isa => 'Int', is => 'rw' );
+
+has 'order_status' => ( isa => 'Str', is => 'rw' );
+
+has 'comments' => ( isa => 'Str', is => 'rw');
+
+has 'create_date' => ( isa => 'Str', is => 'rw');
+
+has 'completion_date' => ( isa => 'Str', is => 'rw');
+
+has 'batches' => ( isa => 'Ref', is => 'rw', default => sub { return []; } );
+
+
+sub BUILD {
+    my $self = shift;
+    my $args = shift;
+    my $people_schema = $self->people_schema();
+
+    if (! $args->{sp_order_id}) {
+	print STDERR "Creating empty object...\n";
+	return $self;
+    }
+
+    my $row = $people_schema->resultset('SpOrder')->find( { sp_order_id => $args->{sp_order_id} } );
+
+    if (!$row) {
+	die "The database has no order entry with id $args->{sp_order_id}";
+    }
+
+    $self->order_from_id($row->order_from_id);
+    $self->order_to_id($row->order_to_id);
+    $self->create_date($row->create_date);
+
+}
+
+
+sub store {
+    my $self = shift;
+    my %data = (
+	order_status => $self->order_status(),
+	order_from_id => $self->order_from_id(),
+	order_to_id => $self->order_to_id(),
+	comments => $self->comments(),
+    create_date => $self->create_date(),
+    completion_date => $self->completion_date(),
+	);
+
+    if ($self->sp_order_id()) { $data{sp_order_id} = $self->sp_order_id(); }
+
+    my $rs = $self->people_schema()->resultset('SpOrder');
+
+    my $row = $rs->update_or_create( \%data );
+
+    return $row->sp_order_id();
+}
+
+
+sub get_orders_from_person_id {
+    my $self = shift;
+    my $people_schema = $self->people_schema();
+    my $person_id = $self->order_from_id();
+    my $dbh = $self->dbh();
+
+    my $order_rs = $people_schema->resultset('SpOrder')->search( { order_from_id => $person_id } );
```

Comment:
DBIx::Class::Schema::source(): Can't find source for SpOrder
Was there a pull request to bio chado schema to add this?